src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4056,5 +4074,47 @@ def parse_lsof(lsof, dirs):
     return list(pids.items())
 
 
+def parse_fstat(fstat, dirs):
+
+    # fstat output is separated by newlines
+    fstat = fstat.split('\n')
+
+    # first line is the column headers
+    fstat.pop(0)
+
+    def keep(fields):
+
+        # drop empty lines and/or lines without a 5th column
+        # because that's the path associated to the process
+        if len(fields) < 5:
+            return False
+
+        # drop the g_eli/g_mirror information since it's not needed
+        if fields[1].startswith(('g_eli', 'g_mirror')):
+            return False
+
+        # only return lines that have `/` in the path information since
+        # paths can be stream/socket/pipe types
+        return '/' in fields[4]
+
+    pids = {}
+    pid = command = None
+    for line in filter(keep, map(str.split, fstat)):
+        pid = command = None
+
+        try:
+            pid = int(line[2])
+        except ValueError:
+            pass
```

Comment:
Didnt you mean `continue`?

Reply:
I tried to mimic the exact behavior of the parse_load method.

Reply:
Well behavior there seems wrong.

No point in continuing if `pids[pid] = command` will never be reached if we can't parse the pid. Its just wasting cpu cycles.

Reply:
Agree, I'll fix.